Amplitude/Amplitude.m

```diff
@@ -1185,9 +1173,23 @@ - (void)enterBackground
         [self->_dbHelper insertOrReplaceKeyLongValue:OPT_OUT value:[NSNumber numberWithBool:self->_optOut]];
         [self->_dbHelper insertOrReplaceKeyLongValue:PREVIOUS_SESSION_ID value:[NSNumber numberWithLongLong:self->_sessionId]];
         [self->_dbHelper insertOrReplaceKeyLongValue:PREVIOUS_SESSION_TIME value:[NSNumber numberWithLongLong:self->_lastEventTime]];
+
+        [self endBackgroundTaskIfNeeded];
```

Comment:
yep, i've just checked on demo app. with real device(not simulator), flushing does not have time to complete, because background task ends right on start of flushing